frontend/src/features/projects/common/projectConverter.ts

```diff
@@ -74,7 +74,8 @@ export const toFlatProject = (project?: IApiProject) => {
     };
     return property;
   });
-  const flatProject: IProject = { ...project, properties: flatProperties };
+  //always copy the project values over initial values, this ensures that formik's requirement of non-undefined fields is satisfied.
+  const flatProject: IProject = { ...initialValues, ...project, properties: flatProperties };
```

Comment:
This is what I mentioned above. Start with the default empty string values and then overwrite with the actual project values to ensure no undefined values leak into the forms.